### DISCUSSION TITLE
"Only applies on the way down" check for "Time Average" custom sensor.

### DISCUSSION THREAD

**Discussion by Cantejito** [Rem0o/FanControl.Releases#2385] [Ideas]
- **Idea**
Add "only applies on the way down" to the "Time Average" custom sensor.

- **Why?**
You could make a curve which would allow fast reaction to temperature rises at fluctuating loads without having to listen to the fans constantly going up and down. Fans would maintain a high profile until they slow down gradually due to the temperature drop.

- **How?**
The sensor will check if the current temperature is higher than it was X time ago. If true, it will refesh to the current temperature.

- **Notes**
I know it may have flaws, but I think is better to have the choice.

**Comment by Rem0o** [comment]
What's the problem with normal the current method of  hysteresis + high response time + low step-down? It does what you describe, fast reaction to temp rise, slow reaction to temp drops for a more gradual slow down.

**Reply by Cantejito** [reply]
The problem with that is that it only works with fans that match speed and curves, as the fan is getting its "step down" from a percentage that could mean anything in terms of RPM.

I'm currently using a mix of an old fan from an AIO, the ones that came with the case and another one I bought when one died. It would be a pain to even try to get them to respond with the same "sensitivity", but with the proposed solution, the fans are guided by one constant; temperature. 50º will be 50º in my system or yours, but 10% reduction in my fans or yours will not. Not even between my own fans.

The main reason for this idea comes from reading other discussions about the transition to fan RPM and its accuracy (up to 35 rpm iirc). That feature would be in line with those options.

As always, thank you for your job with FanControl. For me is the one and only option in its category. 😊

**Reply by Rem0o** [reply]
Forget the step down%, only focus on the existing response time + hysteresis. 

Let say a RPM mode graph, with 15 second response time with 10 degree hysteresis + hysteresis on the way down checked.  Like this:
![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/1d0f0bd5-1514-4f78-81ed-3603545a7e4a)

How does that not fill in your need? You can hook different types of fan, they will all run at the same speed. There is a fast response time to temp increase, and slow response to temp decrease.

**Reply by Cantejito** [reply]
That's almost what i'm looking for, but is more than good enough.

I was getting confused by the tooltip "Time for the fan curve in seconds (sec) to detect a change in temperature" and I thought that response time was on his own and will affect the curve both ways and will not be affected by the hysteresis option.

Sorry for misunderstanding your tool 😅

**Reply by Rem0o** [reply]
No problem! Best way to figure out the different options is to experiment a bit :)
